vulnfeeds/cpp/main.go

```diff
@@ -140,9 +173,75 @@ func MaybeGetSourceRepoFromDebian(mdir string, pkg string) string {
 	return ""
 }
 
+// Use the GitHub API to query the repository's language metadata to make the determination.
+func InScopeGitHubRepo(repoURL string) bool {
+	// TODO(apollock): Implement
+	return true
+}
+
+// Clone the repo and look for C/C++ files to make the determination.
+func InScopeGitRepo(repoURL string) bool {
+	// TODO(apollock): Implement
+	return true
+}
+
+// Looks at what the repo to determine if it contains code using an in-scope language
+func InScopeRepo(repoURL string) bool {
+	parsedURL, err := url.Parse(repoURL)
+	if err != nil {
+		Logger.Infof("Warning: %s failed to parse, skipping", repoURL)
+		return false
+	}
+
+	switch parsedURL.Hostname() {
+	case "github.com":
+		return InScopeGitHubRepo(repoURL)
+	default:
+		return InScopeGitRepo(repoURL)
+	}
+}
+
+// Takes an NVD CVE record and outputs an OSV file in the specified directory.
+func CVEToOSV(cve cves.CVEItem, directory string) {
+	CPEs := cves.CPEs(cve)
+	CPE, ok := cves.ParseCPE(CPEs[0])
+	if !ok {
+		Logger.Fatalf("Can't generate an OSV record for %s without CPE data", cve.CVE.CVEDataMeta.ID)
+	}
+	versions, _ := cves.ExtractVersion(cve)
+	v, _ := vulns.FromCVE(cve.CVE.CVEDataMeta.ID, cve)
+	pkgInfo := vulns.PackageInfo{
```

Comment:
Let's just not have this package information at all.

Reply:
Done.

Reply:
Oh I remember why I was doing this. AddPkgInfo has a side-effect of adding the Affected field to the vulnerability...

It seems to have a lot of commit-related logic as well. Am I better off modifying it to support the PackageInfo struct not having ecosystem and PURL specified?

Reply:
Yep, let's refactor things here to have the end result we want.

Reply:
Yeah now I have my head around the schema creation, I've managed to get the outcome I needed.